libs/langgraph/src/graph/messages_annotation.ts

```diff
@@ -45,3 +47,69 @@ export const MessagesAnnotation = Annotation.Root({
     default: () => [],
   }),
 });
+
+/**
+ * Prebuilt state object that uses Zod to combine returned messages.
+ * This utility is synonymous with the `MessagesAnnotation` annotation,
+ * but uses Zod as the way to express messages state.
+ *
+ * You can use import and use this prebuilt schema like this:
+ *
+ * @example
+ * ```ts
+ * import { MessagesZodState, StateGraph } from "@langchain/langgraph";
+ *
+ * const graph = new StateGraph(MessagesZodState)
+ *   .addNode(...)
+ *   ...
+ * ```
+ *
+ * Which is equivalent to initializing the schema object manually like this:
+ *
+ * @example
+ * ```ts
+ * import { BaseMessage } from "@langchain/core/messages";
+ * import { Annotation, StateGraph, messagesStateReducer } from "@langchain/langgraph";
+ *
+ * const schema = z.object({
+ *   messages: z
+ *     .array(z.string())
+ *     .default(() => [])
+ *     .langgraph.reducer(
+ *        messagesStateReducer,
+ *        z.union([z.string(), z.array(z.string())])
+ *     ),
+ * });
+ * export const StateAnnotation = Annotation.Root({
+ *   messages: Annotation<BaseMessage[]>({
+ *     reducer: messagesStateReducer,
+ *     default: () => [],
+ *   }),
+ * });
+ *
+ * You can also expand this schema to include other fields and retain the core messages field using native zod methods like `z.intersection()` or `.and()`
+ * @example
+ * ```ts
+ * import { MessagesZodState, StateGraph } from "@langchain/langgraph";
+ *
+ * const schema = MessagesZodState.and(
+ *   z.object({ count: z.number() }),
+ * );
+ *
+ * const graph = new StateGraph(schema)
+ *  .addNode(...)
+ *  ...
+ * ```
+ */
+export const MessagesZodState = z.object({
+  messages: withLangGraph(z.custom<BaseMessage[]>(), {
+    reducer: {
+      schema: z.union([
+        z.custom<BaseMessageLike>(),
+        z.array(z.custom<BaseMessageLike>()),
```

Comment:
Should we do `z.custom<Messages>` directly instead, given the presence of `z.custom`?